### PR TITLE
Support of polyfill on non-browser non-node.js environments

### PR DESCRIPTION
**Pull request by kt3k**
Hi, thanks for the great library.

I always prefer using this lib through polyfilled style, but I found that in some environments, such as titanium mobile, the current polyfill doesn't work.
After searching a bit on stackoverflow I found a generic way to access the global object, then I suggest this change.

see http://stackoverflow.com/a/3277192/1185492

Thanks,

### CODE REVIEW

File: lib/es6-promise/polyfill.js [hunk 1]
```diff
@@ -2,7 +2,7 @@
 import Promise from './promise';
 
 export default function polyfill() {
-  var local = (typeof global !== 'undefined') ? global : self;
+  var local = Function('return this;')();
```
Comment (stefanpenner) [inline lib/es6-promise/polyfill.js]: this wont work with strict CSP as it is a form of eval

Comment (kt3k) [inline lib/es6-promise/polyfill.js]: How about the following?

```
var local = (typeof global !== 'undefined') ? global :
            (typeof window !== 'undefined' && window.document) ? window : Function('return this;')();
```

It must work on eval-disabled browsers, and also supports generic environment.

Comment (stefanpenner) [inline lib/es6-promise/polyfill.js]: i believe some of these environments disallow the statement all together. Is their no way to just use `this` without the eval in that context?

**Comment by kt3k** [comment]
Now I think the completely generic polyfill is difficult, then I propose a modfied version of this.

**Pull request closed without merging**